lib/src/fixed_datetime_formatter.dart

```diff
@@ -0,0 +1,372 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// A formatter and parser for [DateTime] in a fixed format [String] pattern.
+///
+/// For example, calling
+/// `FixedDateTimeCodec('YYYYMMDDhhmmss').decodeToLocal('19960425050322')` has
+/// the same result as calling `DateTime(1996, 4, 25, 5, 3, 22)`.
+///
+/// The allowed characters are
+/// * `Y`	for “calendar year”
+/// * `M`	for “calendar month”
+/// * `D`	for “calendar day”
+/// * `E`	for “decade”
+/// * `C`	for “century”
+/// * `h`	for “clock hour”
+/// * `m`	for “clock minute”
+/// * `s`	for “clock second”
+/// * `S`	for “fractional clock second”
+///
+/// Note: Negative years are not supported.
+///
+/// Non-allowed characters in the format [pattern] are included when decoding a
+/// string, in this case `YYYY kiwi MM` is the same format string as
+/// `YYYY------MM`. When encoding a [DateTime], the non-format characters are in
+/// the output verbatim.
+///
+/// Note: this class differs from [DateFormat] in that here, the characters are
```

Comment:
Hmm. Keeps feeling slightly off to me. Maybe "the character count is interpreted literally".

Also the example might be more illustrative if it was for `encode`. Can do both, by adding:

> , and the format string `YYY` used to encode the year `1996` would output only the three 
> digits `996`.